### PR TITLE
Feature/ti2023 38/join event(Backend)

### PR DESCRIPTION
**Pull request by bObXm**
Join event with send email to creator of the event when someone joins.

User can join only one position per event.
On each join the number of spots for that position decreases.

### CODE REVIEW

**Review by HellHound97 — APPROVED** [review]

File: EventPlannerAPI/BusinessLayer/Services/EventService.cs [hunk 1]
```diff
@@ -173,5 +177,57 @@ private async Task<string> ValidateUpdateEventDtoAsync(Guid sportTypeId, UpdateE
...
+                }
+
+                var userDetails=await _userRepository.GetUserProfileDetailsAsync(userId);
+                if (userDetails == null)
+                {
+                    _logger.Error($"User details for user id {userId} not found.");
+                    throw new KeyNotFoundException("User details not found");
+                }
+
+                var profileLink = $"http://localhost:3000/profile/{userId}";
```
Comment (alexandratodea) [inline EventPlannerAPI/BusinessLayer/Services/EventService.cs]: this shouldn't be here

File: EventPlannerAPI/DataAccessLayer/Repositories/EventRepository.cs [hunk 2]
```diff
@@ -86,5 +88,44 @@ public async Task<bool> PositionBelongsToSportTypeAsync(Guid positionId, Guid sp
...
+                }
+                await _eventPlannerContext.Participants.AddAsync(participant);
+                await _eventPlannerContext.SaveChangesAsync();
+            }
+            catch (Exception ex)
+            {
+                throw;
+            };
+
+            return "User joined the event successfully.";
```
Comment (alexandratodea) [inline EventPlannerAPI/DataAccessLayer/Repositories/EventRepository.cs]: this method is async

File: EventPlannerAPI/EventPlannerAPI/Controllers/AdminController.cs [hunk 3]
```diff
@@ -12,7 +12,7 @@ namespace EventPlannerAPI.Controllers
 {
     [Route("api/[controller]")]
     [ApiController]
-    [Authorize(Roles = RoleConstants.ADMIN_ROLE)]
+/*    [Authorize(Roles = RoleConstants.ADMIN_ROLE)]*/
```
Comment (alexandratodea) [inline EventPlannerAPI/EventPlannerAPI/Controllers/AdminController.cs]: uncomment

File: EventPlannerAPI/EventPlannerAPI/Controllers/EventController.cs [hunk 4]
```diff
@@ -12,7 +12,7 @@ namespace EventPlannerAPI.Controllers
 {
     [Route("api/[controller]")]
     [ApiController]
-    [Authorize]
+  /*  [Authorize]*/
```
Comment (alexandratodea) [inline EventPlannerAPI/EventPlannerAPI/Controllers/EventController.cs]: uncomment

**Review by alexandratodea — APPROVED** [review]

**Pull request merged**